<a href="https://colab.research.google.com/github/Nitrolego/ml-playground/blob/main/learning.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

#Setup

In [3]:
#asserts python version
import sys
assert sys.version_info >= (3, 7)

#imports scikit-learn
from packaging import version
import sklearn

assert version.parse(sklearn.__version__) >= version.parse("1.0.1")

In [24]:
from sklearn.datasets import load_wine

wine = load_wine()

In [23]:
from sklearn.datasets import load_wine

#loads wine dataset as a pandas dataframe
winex, winey = load_wine(return_X_y=True, as_frame=True)

winex.shape
#data needs to be a data matrix, in the dataframe to be able to do head/info/etc operations

(178, 13)

In [25]:
winex.info

<bound method DataFrame.info of      alcohol  malic_acid   ash  alcalinity_of_ash  magnesium  total_phenols  \
0      14.23        1.71  2.43               15.6      127.0           2.80   
1      13.20        1.78  2.14               11.2      100.0           2.65   
2      13.16        2.36  2.67               18.6      101.0           2.80   
3      14.37        1.95  2.50               16.8      113.0           3.85   
4      13.24        2.59  2.87               21.0      118.0           2.80   
..       ...         ...   ...                ...        ...            ...   
173    13.71        5.65  2.45               20.5       95.0           1.68   
174    13.40        3.91  2.48               23.0      102.0           1.80   
175    13.27        4.28  2.26               20.0      120.0           1.59   
176    13.17        2.59  2.37               20.0      120.0           1.65   
177    14.13        4.10  2.74               24.5       96.0           2.05   

     flavanoids  no

In [28]:
winey.info

<bound method Series.info of 0      0
1      0
2      0
3      0
4      0
      ..
173    2
174    2
175    2
176    2
177    2
Name: target, Length: 178, dtype: int64>

In [31]:
wine.target_names

array(['class_0', 'class_1', 'class_2'], dtype='<U7')

# Splitting training and test set

In [26]:
from sklearn.model_selection import train_test_split

x_train, x_test , y_train, y_test= train_test_split(winex, winey, random_state=42)

x train is now 133, total data set, 178

x test is 178-133??
yes it is,

shape() to tell.

In [22]:
x_train.shape

(133, 13)

In [21]:
x_test.shape

(45, 13)

# classification

In [37]:
from sklearn.linear_model import SGDClassifier

clf = SGDClassifier(loss="hinge", penalty="l2")
clf.fit(x_train, y_train)

SGDClassifier()

what class labels are for x?

In [35]:
clf.predict(x_test)

array([0, 0, 1, 0, 1, 0, 1, 0, 1, 0, 0, 1, 0, 1, 0, 1, 1, 0, 0, 1, 0, 1,
       1, 1, 1, 1, 1, 1, 1, 0, 0, 1, 0, 0, 0, 0, 1, 1, 0, 1, 0, 1, 1, 1,
       1])

predicted!

next step, calculate accuracy.

# evaluation of performance

In [42]:
from sklearn.model_selection import cross_val_score

cross_val_score(clf, x_test, y_test)

array([0.44444444, 0.66666667, 0.44444444, 0.33333333, 0.77777778])

grid search cv?